crates/ra_hir/src/ty.rs

```diff
@@ -449,6 +449,14 @@ pub fn type_for_enum(db: &impl HirDatabase, s: Enum) -> Cancelable<Ty> {
     })
 }
 
+pub fn type_for_enum_variant(db: &impl HirDatabase, ev: EnumVariant) -> Cancelable<Ty> {
+    let enum_def = ev.enum_def_id().resolve(db)?;
+    match enum_def {
+        Def::Enum(e) => type_for_enum(db, e),
+        _ => unreachable!("expected enum as enum variant's ancestor"),
```

Comment:
you wouldn't need this match here if `EnumVariant` just had an `enum` method directly returning the `Enum` :)

Reply:
yup - done!